File: language-support/go/template.go
/*
Go Language Support Template
For MASTER-WORKFLOW v3.0
*/

package main

import (
	"encoding/json"
	"fmt"
	"log"
)

// LanguageConfig represents the Go language configuration
type LanguageConfig struct {
	Language   string                 `json:"language"`
	Extensions []string               `json:"extensions"`
	BuildTools BuildTools             `json:"buildTools"`
	Frameworks map[string][]string    `json:"frameworks"`
	Testing    TestingConfig          `json:"testing"`
	Linting    LintingConfig          `json:"linting"`
	Patterns   map[string]interface{} `json:"patterns"`
	DevServer  DevServerConfig        `json:"devServer"`
	Deployment DeploymentConfig       `json:"deployment"`
}

// BuildTools configuration for Go
type BuildTools struct {
	PackageManager       string   `json:"packageManager"`
	ModuleSystem         string   `json:"moduleSystem"`
	BuildCommand         string   `json:"buildCommand"`
	TestCommand          string   `json:"testCommand"`
	LintCommand          string   `json:"lintCommand"`
	FormatCommand        string   `json:"formatCommand"`
	VendorCommand        string   `json:"vendorCommand"`
	ModInit              string   `json:"modInit"`
	AlternativeManagers  []string `json:"alternativeManagers"`
}

// TestingConfig for Go testing frameworks
type TestingConfig struct {
	Unit       []string `json:"unit"`
	Benchmark  []string `json:"benchmark"`
	Coverage   []string `json:"coverage"`
	E2E        []string `json:"e2e"`
	ConfigFile string   `json:"configFile"`
}

// LintingConfig for Go linting and formatting
type LintingConfig struct {
	Linter          string `json:"linter"`
	Config          string `json:"config"`
	Formatter       string `json:"formatter"`
	StaticAnalysis  []string `json:"staticAnalysis"`
	SecurityScanner []string `json:"securityScanner"`
}

// DevServerConfig for Go development
type DevServerConfig struct {
	Command     string `json:"command"`
	DefaultPort int    `json:"defaultPort"`
	HotReload   bool   `json:"hotReload"`
	WatchTool   string `json:"watchTool"`
}

// DeploymentConfig for Go deployment options
type DeploymentConfig struct {
	Platforms        []string `json:"platforms"`
	Containerization string   `json:"containerization"`
	CI               []string `json:"ci"`
	CloudNative      []string `json:"cloudNative"`
}

// GetGoConfig returns the complete Go language configuration
func GetGoConfig() *LanguageConfig {
	return &LanguageConfig{
		Language:   "go",
		Extensions: []string{".go", ".mod", ".sum"},
		BuildTools: BuildTools{
			PackageManager:      "go modules",
			ModuleSystem:        "go.mod",
			BuildCommand:        "go build",
			TestCommand:         "go test ./...",
			LintCommand:         "golangci-lint run",
			FormatCommand:       "gofmt -s -w .",
			VendorCommand:       "go mod vendor",
			ModInit:             "go mod init",
			AlternativeManagers: []string{"dep", "glide", "godep"},
		},
		Frameworks: map[string][]string{
			"web":        {"Gin", "Echo", "Fiber", "Chi", "Gorilla Mux", "Buffalo"},
			"grpc":       {"gRPC-Go", "Twirp", "Connect"},
			"orm":        {"GORM", "Ent", "SQLBoiler", "Squirrel"},
			"testing":    {"Testify", "GoConvey", "Ginkgo", "Gomega"},
			"cli":        {"Cobra", "Urfave CLI", "Kong", "Kingpin"},
			"microservices": {"Go-kit", "Micro", "Kratos", "Go-chassis"},
		},
		Testing: TestingConfig{
			Unit:       []string{"testing", "testify/assert", "testify/mock", "testify/suite"},
			Benchmark:  []string{"testing.B", "benchstat", "gobench"},
			Coverage:   []string{"go test -cover", "gocov", "gcov2lcov"},
			E2E:        []string{"Selenium", "Agouti", "Chromedp"},
			ConfigFile: ".golangci.yml",
		},
		Linting: LintingConfig{
			Linter:          "golangci-lint",
			Config:          ".golangci.yml",
			Formatter:       "gofmt",
			StaticAnalysis:  []string{"go vet", "staticcheck", "gosec", "ineffassign"},
			SecurityScanner: []string{"gosec", "snyk", "nancy"},
		},
		Patterns: map[string]interface{}{
			"concurrency":     "goroutines and channels",
			"errorHandling":   "explicit error returns",
			"interfaceDesign": "small interfaces",
			"dependency":      "dependency injection",
			"configuration":   "environment variables and config structs",
		},
		DevServer: DevServerConfig{
			Command:     "go run main.go",
			DefaultPort: 8080,
			HotReload:   false,
			WatchTool:   "air",
		},
		Deployment: DeploymentConfig{
			Platforms:        []string{"Google Cloud", "AWS", "Azure", "Digital Ocean", "Heroku"},
			Containerization: "Docker",
			CI:               []string{"GitHub Actions", "GitLab CI", "Jenkins", "CircleCI"},
			CloudNative:      []string{"Kubernetes", "Docker Swarm", "Nomad"},
		},
	}
}

func main() {
	config := GetGoConfig()
	jsonData, err := json.MarshalIndent(config, "", "  ")
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(jsonData))
}

File: language-support/java/template.java
/*
 * Java Language Support Template
 * For MASTER-WORKFLOW v3.0
 */

package com.masterworkflow.language.java;

import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

/**
 * Java Language Configuration Template
 * Provides comprehensive configuration for Java projects
 */
public class JavaLanguageConfig {
    
    private final String language = "java";
    private final List<String> extensions = Arrays.asList(".java", ".class", ".jar", ".war");
    private final BuildTools buildTools;
    private final Map<String, List<String>> frameworks;
    private final TestingConfig testing;
    private final LintingConfig linting;
    private final Map<String, Object> patterns;
    private final DevServerConfig devServer;
    private final DeploymentConfig deployment;
    private final JavaSpecificConfig javaConfig;

    public JavaLanguageConfig() {
        this.buildTools = new BuildTools();
        this.frameworks = initializeFrameworks();
        this.testing = new TestingConfig();
        this.linting = new LintingConfig();
        this.patterns = initializePatterns();
        this.devServer = new DevServerConfig();
        this.deployment = new DeploymentConfig();
        this.javaConfig = new JavaSpecificConfig();
    }

    public static class BuildTools {
        @JsonProperty("packageManager")
        private final String packageManager = "Maven";
        
        @JsonProperty("alternativeManagers")
        private final List<String> alternativeManagers = Arrays.asList("Gradle", "SBT", "Ant", "Bazel");
        
        @JsonProperty("buildCommand")
        private final String buildCommand = "mvn clean compile";
        
        @JsonProperty("testCommand")
        private final String testCommand = "mvn test";
        
        @JsonProperty("lintCommand")
        private final String lintCommand = "mvn checkstyle:check";
        
        @JsonProperty("formatCommand")
        private final String formatCommand = "mvn spotless:apply";
        
        @JsonProperty("packageCommand")
        private final String packageCommand = "mvn package";
        
        @JsonProperty("runCommand")
        private final String runCommand = "mvn exec:java";

        // Getters
        public String getPackageManager() { return packageManager; }
        public List<String> getAlternativeManagers() { return alternativeManagers; }
        public String getBuildCommand() { return buildCommand; }
        public String getTestCommand() { return testCommand; }
        public String getLintCommand() { return lintCommand; }
        public String getFormatCommand() { return formatCommand; }
        public String getPackageCommand() { return packageCommand; }
        public String getRunCommand() { return runCommand; }
    }

    public static class TestingConfig {
        @JsonProperty("unit")
        private final List<String> unit = Arrays.asList("JUnit 5", "TestNG", "Spock", "JUnit 4");
        
        @JsonProperty("integration")
        private final List<String> integration = Arrays.asList("Spring Boot Test", "Testcontainers", "WireMock");
        
        @JsonProperty("mocking")
        private final List<String> mocking = Arrays.asList("Mockito", "PowerMock", "EasyMock");
        
        @JsonProperty("coverage")
        private final List<String> coverage = Arrays.asList("JaCoCo", "Cobertura", "Clover");
        
        @JsonProperty("bdd")
        private final List<String> bdd = Arrays.asList("Cucumber", "Spock", "JBehave");
        
        @JsonProperty("performance")
        private final List<String> performance = Arrays.asList("JMH", "Gatling", "JMeter");

        // Getters
        public List<String> getUnit() { return unit; }
        public List<String> getIntegration() { return integration; }
        public List<String> getMocking() { return mocking; }
        public List<String> getCoverage() { return coverage; }
        public List<String> getBdd() { return bdd; }
        public List<String> getPerformance() { return performance; }
    }

    public static class LintingConfig {
        @JsonProperty("linter")
        private final String linter = "Checkstyle";
        
        @JsonProperty("config")
        private final String config = "checkstyle.xml";
        
        @JsonProperty("formatter")
        private final String formatter = "Spotless";
        
        @JsonProperty("staticAnalysis")
        private final List<String> staticAnalysis = Arrays.asList("SpotBugs", "PMD", "SonarQube", "Error Prone");
        
        @JsonProperty("securityScanner")
        private final List<String> securityScanner = Arrays.asList("OWASP Dependency Check", "Snyk", "Veracode");

        // Getters
        public String getLinter() { return linter; }
        public String getConfig() { return config; }
        public String getFormatter() { return formatter; }
        public List<String> getStaticAnalysis() { return staticAnalysis; }
        public List<String> getSecurityScanner() { return securityScanner; }
    }

    public static class DevServerConfig {
        @JsonProperty("command")
        private final String command = "mvn spring-boot:run";
        
        @JsonProperty("defaultPort")
        private final int defaultPort = 8080;
        
        @JsonProperty("hotReload")
        private final boolean hotReload = true;
        
        @JsonProperty("devTools")
        private final List<String> devTools = Arrays.asList("Spring DevTools", "JRebel", "HotSwap Agent");

        // Getters
        public String getCommand() { return command; }
        public int getDefaultPort() { return defaultPort; }
        public boolean isHotReload() { return hotReload; }
        public List<String> getDevTools() { return devTools; }
    }

    public static class DeploymentConfig {
        @JsonProperty("platforms")
        private final List<String> platforms = Arrays.asList("AWS", "Google Cloud", "Azure", "Heroku", "Railway");
        
        @JsonProperty("containerization")
        private final String containerization = "Docker";
        
        @JsonProperty("ci")
        private final List<String> ci = Arrays.asList("Jenkins", "GitHub Actions", "GitLab CI", "CircleCI", "Azure DevOps");
        
        @JsonProperty("applicationServers")
        private final List<String> applicationServers = Arrays.asList("Tomcat", "Jetty", "Undertow", "WebLogic", "WebSphere");
        
        @JsonProperty("cloudNative")
        private final List<String> cloudNative = Arrays.asList("Kubernetes", "OpenShift", "Cloud Foundry");

        // Getters
        public List<String> getPlatforms() { return platforms; }
        public String getContainerization() { return containerization; }
        public List<String> getCi() { return ci; }
        public List<String> getApplicationServers() { return applicationServers; }
        public List<String> getCloudNative() { return cloudNative; }
    }

    public static class JavaSpecificConfig {
        @JsonProperty("version")
        private final String version = "17+";
        
        @JsonProperty("lts")
        private final List<String> lts = Arrays.asList("8", "11", "17", "21");
        
        @JsonProperty("jvm")
        private final List<String> jvm = Arrays.asList("HotSpot", "OpenJ9", "GraalVM", "Azul Zulu");
        
        @JsonProperty("features")
        private final Map<String, Boolean> features = Map.of(
            "lambdas", true,
            "streams", true,
            "modules", true,
            "records", true,
            "textBlocks", true,
            "patternMatching", true,
            "virtualThreads", true
        );

        // Getters
        public String getVersion() { return version; }
        public List<String> getLts() { return lts; }
        public List<String> getJvm() { return jvm; }
        public Map<String, Boolean> getFeatures() { return features; }
    }

    private Map<String, List<String>> initializeFrameworks() {
        Map<String, List<String>> frameworks = new HashMap<>();
        frameworks.put("web", Arrays.asList("Spring Boot", "Spring MVC", "Jersey", "Dropwizard", "Spark Java", "Vert.x"));
        frameworks.put("enterprise", Arrays.asList("Spring Framework", "Jakarta EE", "Micronaut", "Quarkus", "Helidon"));
        frameworks.put("microservices", Arrays.asList("Spring Cloud", "Micronaut", "Quarkus", "Helidon", "Dropwizard"));
        frameworks.put("orm", Arrays.asList("Hibernate", "JPA", "MyBatis", "JOOQ", "Spring Data"));
        frameworks.put("security", Arrays.asList("Spring Security", "Apache Shiro", "JAAS", "Keycloak"));
        frameworks.put("messaging", Arrays.asList("Spring JMS", "Apache Camel", "RabbitMQ", "Apache Kafka"));
        frameworks.put("reactive", Arrays.asList("Project Reactor", "RxJava", "Akka", "Vert.x"));
        return frameworks;
    }

    private Map<String, Object> initializePatterns() {
        Map<String, Object> patterns = new HashMap<>();
        patterns.put("designPatterns", Arrays.asList("Singleton", "Factory", "Observer", "Strategy", "Command"));
        patterns.put("architecture", Arrays.asList("MVC", "Layered", "Microservices", "Event-Driven", "CQRS"));
        patterns.put("concurrency", "Thread pools, Executors, CompletableFuture");
        patterns.put("dependencyInjection", "Constructor injection, Field injection, Setter injection");
        patterns.put("errorHandling", "Checked exceptions, Runtime exceptions, Try-with-resources");
        return patterns;
    }

    /**
     * Get the complete configuration as a JSON string
     */
    public String toJson() throws JsonProcessingException {
        ObjectMapper mapper = new ObjectMapper();
        return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
    }

    /**
     * Validate the configuration
     */
    public boolean validateConfig() {
        return language.equals("java") && 
               extensions.contains(".java") && 
               buildTools != null && 
               frameworks != null;
    }

    /**
     * Get Maven project template
     */
    public String getMavenTemplate() {
        return """
            <?xml version="1.0" encoding="UTF-8"?>
            <project xmlns="http://maven.apache.org/POM/4.0.0"
                     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                     xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 
                     http://maven.apache.org/xsd/maven-4.0.0.xsd">
                <modelVersion>4.0.0</modelVersion>
                
                <groupId>com.example</groupId>
                <artifactId>my-project</artifactId>
                <version>1.0.0</version>
                <packaging>jar</packaging>
                
                <properties>
                    <maven.compiler.source>17</maven.compiler.source>
                    <maven.compiler.target>17</maven.compiler.target>
                    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
                    <junit.version>5.9.2</junit.version>
                    <spring.boot.version>3.1.0</spring.boot.version>
                </properties>
                
                <dependencies>
                    <dependency>
                        <groupId>org.springframework.boot</groupId>
                        <artifactId>spring-boot-starter-web</artifactId>
                        <version>${spring.boot.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>org.junit.jupiter</groupId>
                        <artifactId>junit-jupiter</artifactId>
                        <version>${junit.version}</version>
                        <scope>test</scope>
                    </dependency>
                </dependencies>
                
                <build>
                    <plugins>
                        <plugin>
                            <groupId>org.springframework.boot</groupId>
                            <artifactId>spring-boot-maven-plugin</artifactId>
                            <version>${spring.boot.version}</version>
                        </plugin>
                        <plugin>
                            <groupId>org.apache.maven.plugins</groupId>
                            <artifactId>maven-surefire-plugin</artifactId>
                            <version>3.1.2</version>
                        </plugin>
                    </plugins>
                </build>
            </project>
            """;
    }

    // Getters
    public String getLanguage() { return language; }
    public List<String> getExtensions() { return extensions; }
    public BuildTools getBuildTools() { return buildTools; }
    public Map<String, List<String>> getFrameworks() { return frameworks; }
    public TestingConfig getTesting() { return testing; }
    public LintingConfig getLinting() { return linting; }
    public Map<String, Object> getPatterns() { return patterns; }
    public DevServerConfig getDevServer() { return devServer; }
    public DeploymentConfig getDeployment() { return deployment; }
    public JavaSpecificConfig getJavaConfig() { return javaConfig; }

    /**
     * Main method for demonstration
     */
    public static void main(String[] args) {
        try {
            JavaLanguageConfig config = new JavaLanguageConfig();
            System.out.println("Java Language Configuration:");
            System.out.println(config.toJson());
            
            System.out.println("\nValidation Result: " + config.validateConfig());
            
            System.out.println("\nMaven Template:");
            System.out.println(config.getMavenTemplate());
            
        } catch (JsonProcessingException e) {
            System.err.println("Error generating JSON: " + e.getMessage());
        }
    }
}